lib/os_io.c

```diff
@@ -37,6 +37,199 @@ ___io_module ___io_mod =
 
 /*---------------------------------------------------------------------------*/
 
+/* poll dynamic fdset memory management. */
+
+#ifdef USE_select_or_poll
+
+___HIDDEN int ___fdset_heap_overflow;
```

Comment:
Exactly. It is ok however if some processors have a fdset bigger than the VM size.